src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,745 @@
+          );
+          return Optional.of(stream);
+        }
+      );
+  }
+
+  @BeforeEach
+  public void setup() throws Exception {
+    usageAnalysisController = new UsageAnalysisController();
+    MockitoAnnotations.initMocks(this);
+    this.mockS3ObjectResponse(data);
+    this.restMockMvc =
+      MockMvcBuilders.standaloneSetup(usageAnalysisController).build();
+  }
+
+  @Test
+  public void shouldGetUsageSummaryResources() throws Exception {
+    String url = "/api/usage/summary/resources";
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageResources() throws Exception {
+    String url = "/api/usage/resources?endpoint=" + data.sampleUrls[0];
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      // strict checking enforces the same order on arrays
+      // we need to use strict checks to make sure we are not missing
+      // any data in our asserts
+      // if you have issues with arrays make sure you have the correct order
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageSummaryUsers() throws Exception {
+    String url = "/api/usage/summary/users";
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageSummaryUsersByCompanyId() throws Exception {
+    String url =
+      "/api/usage/summary/users?companyId=" +
+      data.userDtos.get(0).getCompany().getId();
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageUsersById() throws Exception {
+    String url = "/api/usage/users/" + data.users.get(0).getId();
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  private static class MockS3Data {
+
+    private static class Tuple<X, Y> {
+      public final X first;
+      public final Y second;
```

Comment:
Give some meaningful names here? It's used by this test anyway, may as well just use lastKey and jsonObject?

Reply:
It's use by the logic that creates the mock json file, but I'll change it to `KeyValuePair`